test/ioc/IocTest.java

```diff
@@ -0,0 +1,49 @@
+package ioc;
+
+
+import exceptions.ioc.IocDuplicatedKeyException;
+import exceptions.ioc.IocKeyNotFoundException;
+import exceptions.ioc.IocKeyNullPointerException;
+import exceptions.ioc.IocValueNullPointerException;
+import org.junit.Before;
+import org.junit.Test;
+
+public class IocTest {
+  private Ioc ioc = null;
+
+  @Before
+  public void setup() {
+    ioc = Ioc.getInstance();
+    ioc.register("Testing", "Hello");
+    ioc.register("Ariel", "Hernandez");
+    ioc.register("Diana", "Monegro");
+    ioc.register("Republica", "Dominicana");
+  }
+
+  @Test(expected = IocKeyNullPointerException.class)
+  public void register_ShouldThrowIocKeyNullPointerException() {
+    ioc.register(null, null);
+    ioc.register(null, "ValueTest1");
```

Comment:
This statement will never happen if the first throws an exception. Create different test in order to test this scenario